node/core/backing/src/lib.rs

```diff
@@ -347,36 +348,38 @@ impl CandidateBackingJob {
 		let candidate_hash = candidate.hash();
 
 		let statement = match valid {
-			ValidationResult::Valid(outputs, validation_data) => {
+			ValidationResult::Valid(commitments, validation_data) => {
 				// make PoV available for later distribution. Send data to the availability
 				// store to keep. Sign and dispatch `valid` statement to network if we
 				// have not seconded the given candidate.
 				//
 				// If the commitments hash produced by validation is not the same as given by
 				// the collator, do not make available and report the collator.
-				let commitments_check = self.make_pov_available(
-					pov,
-					candidate_hash,
-					validation_data,
-					outputs,
-					|commitments| if commitments.hash() == candidate.commitments_hash {
-						Ok(CommittedCandidateReceipt {
-							descriptor: candidate.descriptor().clone(),
-							commitments,
-						})
-					} else {
-						Err(())
-					},
-				).await?;
-
-				match commitments_check {
-					Ok(candidate) => {
-						self.issued_statements.insert(candidate_hash);
-						Some(Statement::Seconded(candidate))
-					}
-					Err(()) => {
-						self.issue_candidate_invalid_message(candidate.clone()).await?;
-						None
+				if candidate.commitments_hash != commitments.hash() {
+					self.issue_candidate_invalid_message(candidate.clone()).await?;
+					None
+				} else {
+					let erasure_valid = self.make_pov_available(
+						pov,
+						candidate_hash,
+						validation_data,
+						candidate.descriptor.erasure_root,
+					).await?;
+
+					match erasure_valid {
+						Ok(()) => {
+							let candidate = CommittedCandidateReceipt {
+								descriptor: candidate.descriptor().clone(),
+								commitments,
+							};
+
+							self.issued_statements.insert(candidate_hash);
+							Some(Statement::Seconded(candidate))
+						}
+						Err(InvalidErasureRoot) => {
+							self.issue_candidate_invalid_message(candidate.clone()).await?;
```

Comment:
It looks like we don't use `candidate` past this point, so we shouldn't need to `clone` it.

Reply:
```suggestion
							self.issue_candidate_invalid_message(candidate).await?;
```